### PR TITLE
feat(misc): add pr-release workflow

### PR DESCRIPTION
**Pull request by mandarini**
with the help and guidance of @JamesHenry 

TODO:
- [x] test it and make sure it works
- [x] add action to create comment under PR with release info
- [x] add action to create comment under PR if release failed
- [x] Updated PR template to include a comment informing contributor that they can request PR release
- [x] Update contributing with "pr release" section under the Submit PR section
- [x] Make sure forks work - so get fork repo name and checkout that
- [x] remove force-dry-run from workflow

Example comment in this PR: https://github.com/nrwl/nx/pull/22093#issuecomment-1981113184

Example comment in PR from forked repo: https://github.com/nrwl/nx/pull/22179#issuecomment-1981184268

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -254,23 +313,64 @@ jobs:
...
+              DRY_RUN="--dry-run"
+              VERSION="0.0.0-dry-run.0"
+            fi
           else
             DRY_RUN=""
           fi
-          pnpm nx-release --local=false $VERSION $DRY_RUN
+          echo "Version set to: $VERSION"
+          echo "DRY_RUN set to: $DRY_RUN"
+          pnpm nx-release --local=false $VERSION --dry-run
```
Comment (mandarini) [inline .github/workflows/publish.yml:373]: If we are to merge this, the `--dry-run` needs to be replaced with `$DRY_RUN`.

**Comment by mandarini** [comment]
Need to fix following:

context: https://github.com/nrwl/nx/actions/runs/8573206007/job/23497902723#logs

* `published with tag "next"`
* docs release step should not have been triggered

**Comment by mandarini** [comment]
Fixed.

Now, I need to make sure the `if: github.event.inputs.pr` check is correctly handled.

**Comment by JamesHenry** [comment]
Closing in favour of #26509 (still keep author credit for @mandarini)

**Pull request closed without merging**